### ISSUE TITLE
[IMPROVEMENT] share manager should respect kubernetesClusterAutoscalerEnabled also

### ISSUE THREAD

**Issue by Forestsoft-de** [longhorn/longhorn#4171]
## Is your improvement request related to a feature? Please describe
Well, I waited a long time for this #2203 feature. Thanks for that.

After i updated to the latest version i get a new error message from cluster autoscaler:
node cannot be removed: longhorn-system/share-manager-pvc-61760353-2a01-.... is not replicated

## Describe the solution you'd like
It should be possible to drain a node by cluster autoscaler automaticly
The controller for share manager should add the annotation to the pod also.

## Describe alternatives you've considered
Only manual pod deletion of the consumed nfs share.

## Additional context
Ive run longhorn manager on dedicated nodes including attached volumes from the cloud provider. Some nodes are auto created by the autoscaler. On each node there a pod gets scheduled which consumes an nfs share the share manager is attached.

**Comment by derekbit** [comment]
For now, the annotation of the share-manager is not set, but it is feasible. But, the potential issue is the service interruption because the share-manager contains the nfs server.

**Comment by derekbit** [comment]
cc @c3y1huang @innobead

**Comment by Forestsoft-de** [comment]
Yeah i get the point but if I configure the cluster autoscaler explicit i should be aware of potential service interruption. :)
